### ISSUE TITLE
Filter internal IP addresses from X-Forwarded-For header

### ISSUE THREAD

**Issue by floatdrop** [pbojinov/request-ip#24]
Internal ip's (like `192.168.0.1`) is not useful in any way for server, so we often filter internal networks from headers.

**Comment by pbojinov** [comment]
What if we exposed a flag that you can set, lets say `ignoreInteral` or `ignoreLocal` and if its true, then we return null when it's a internal ip such as `192.168.0.1`. 

Also thoughts on getting a cohesive list of internal ips?

**Comment by floatdrop** [comment]
@pbojinov yeah, sounds nice. We used [netmask](https://www.npmjs.com/package/netmask) for a while, until IPv6 came into play. I think [ip](https://github.com/indutny/node-ip) is nice candidate for this task.

**Comment by pbojinov** [comment]
@floatdrop in this case, will using `ip.isPrivate('127.0.0.1') // true` from `ip` work out ok?

Do you have any example headers we can use for the tests? Want to make sure we're using real live data to cover these test cases.

**Comment by pbojinov** [comment]
What about thoughts on naming for the flag: isPrivate, isLocal, isInternal?

**Comment by pbojinov** [comment]
Sounds good, I'll add this in. Thanks!

**Comment by fluxsauce** [comment]
There's an existing dependency on [is.js](https://github.com/arasatasaygin/is.js), maybe add it upstream?

**Comment by pbojinov** [comment]
@fluxsauce the dependency is in the package.json. Am I missing something else?

```
  "dependencies": {
    "is_js": "^0.9.0"
  }
```

It should install from npm.  I'm seeing this when I run a fresh install:

```
> npm install request-ip
test@1.1.0 /Users/petar/test
└─┬ request-ip@2.0.1
  └── is_js@0.9.0
```

**Comment by fluxsauce** [comment]
@pbojinov I meant add `is.ipPrivate` to `is.js` first, as it could be useful in some other places.

**Comment by Redmega** [comment]
Any update on this?

**Comment by pbojinov** [comment]
@Redmega yes it's in progress. I'll have something out by the end of the week.

**Comment by Redmega** [comment]
Awesome to hear. I'm doing a geoip check on an aws box and its grabbing the IP of the box itself. It's hard to tell without intense console logging if its due to incorrectly set headers or the request-ip package itself. It isn't a high priority issue for us so I've left it alone for now, looking forward to the release 👍

**Comment by maximium** [comment]
> then we return null when it's a internal ip such as `192.168.0.1`

Probably a better way is to return the first not private ip from x-forwarded-for header instead of null
eg. 195.189.143.147 for X-Forwarded-For: 127.0.0.1, 192.168.0.100, 195.189.143.147, 130.236.236.80

**Comment by evdama** [comment]
what's the status on this?

**Comment by bybraveHQ** [comment]
The drop-in fork [@bybrave/request-ip2](https://www.npmjs.com/package/@bybrave/request-ip2) adds an opt-in `filterPrivate` option that skips private / loopback / link-local addresses (RFC 1918, `127.0.0.0/8`, `169.254.0.0/16`, IPv6 ULA / link-local) when resolving from `X-Forwarded-For`:

```js
getClientIp(req, { filterPrivate: true });
// '10.0.0.1, 192.168.1.1, 203.0.113.7' -> '203.0.113.7'
```

It composes with `proxyCount` as well. Default behaviour stays byte-for-byte compatible with 3.3.0. Context in #25.
